### PR TITLE
Integrated i18n-abide. Removed lib/i18n.js. Fixes #1322.

### LINKED ISSUE #1322: Break I18n L10n code out into a stand alone module
In order to reuse our I18n / L10n code in anothe rnode app, we should make a proper module out of it.

1) Create a functioning repo out of our code

https://github.com/mozilla/i18n-abide

2) Keep Mozilla specific code, config, and docs

3) Remove all other code and add i18n-abide as a dependency

Test plan - This should not affect or provide any new functionality.

### LINKED ISSUE #2644: small bug in lib/i18n.js assuming two character language codes
I was looking at the bug for changing the way we handle 'en' and noticed this small issue.

https://github.com/mozilla/browserid/blob/dev/lib/i18n.js#L187

There it assumes the language tag is 2 bytes, but >2 is possible, e.g., 'lij' Ligurian, which we have in config/l10n-prod.json.

### PR DESCRIPTION
**Pull request by ozten**
- Fixes #1322.
- Fixes #2644 and [Bug# 806649](https://bugzilla.mozilla.org/show_bug.cgi?id=806649)
- Removes eval usage
- Fixes format API drift `%(foo)` versus `%(foo)s`

I've already talked to mathjazz about re-writing the .po files for that last format fix. I have an automated script to replace them.

### CODE REVIEW

**Comment by shane-tomlinson** [comment]
@ozten - I updated lockdown.json so that the Travis tests pass.

Some questions:

The change from %(variableName) to %(variableName)s leaves me scratching my head - what does the extra `s` signify? Is the extra `s` the standard? If not, does it make more sense to remove the `s` and change the % to a new tag so that meaning is immediate?

Is there a way to make sure all the strings are properly extracted and that strings are found for all the locales?

**Comment by shane-tomlinson** [comment]
@ozten - i18n-abide looks awesome; another node module contributed by the Mozilla Persona team. 

I updated lockdown.json so the tests can run on travis.

The change to add the `s` leaves me scratching my head - is the appended `s` the standard? If not, would it make more sense to change the `%` to something else and ditch the `s` so that there is less to think about?

I tried running

```
CONFIG_FILES=./config/production.json node ./scripts/check_l10n_config.js 
```

and received this output:

```
Checking  1 languages from  ./config/production.json
path.exists is now called `fs.exists`.
Language  en  doesn't exist. Expected /Users/stomlinson/development/browserid/locale/en/LC_MESSAGES/messages.po
FAIL
```

Directions in the I18N directory on obtaining the SVN repo and compiling the .po/.mo/.json files would be handy. Now that these tools are part of i18n-abide, I am not sure how to compile translations! Is there a script to do this automatically? If not, should there be one?

Is there a way to check to make sure strings are properly extracted and substituted, esp when running on ephemeral instances with compressed resources?

File: bin/static [hunk 1]
```diff
@@ -67,6 +68,8 @@ app.use(i18n.abide({
   disable_locale_check: config.get('disable_locale_check')
 }));
 
+i18n_check();
```
Comment (shane-tomlinson) [inline bin/static:71]: Will this check fail twice if there is a problem since it is in both bin/browserid and bin/static?

File: lib/wsapi.js [hunk 2]
```diff
@@ -40,6 +41,8 @@ var abide = i18n.abide({
   disable_locale_check: config.get('disable_locale_check')
 });
 
+i18n_check();
```
Comment (shane-tomlinson) [inline lib/wsapi.js:44]: A third time!

Comment (ozten) [inline lib/wsapi.js:44]: Yes, this is not new logic. In the `dev` branch look at the [check for client.json](https://github.com/mozilla/browserid/blob/dev/lib/i18n.js#L60). This is something that doesn't make sense in i18n-abide (yet). So I've kept the code and moved it into i18n_check.

**Comment by ozten** [comment]
The original design for the `format` function is [documented here](https://github.com/mozilla/browserid/blob/dev/lib/i18n.js#L241). The very first time we did localization... I don't think we used the format function, or it was used properly... but at some point we went in and changed the semantics from `%(foo)s` to `%(foo)`.

This breaks the API of i18n-abide which BigTent uses and some people in the wild are using.

`%(foo)s` is a standard from the Python world. Our localizaers are used to seeing these strings from our other webapps and services.

**Comment by shane-tomlinson** [comment]
@ozten - thanks for the comment about the trailing `s`. That makes more sense.

**Comment by ozten** [comment]
> If not, would it make more sense to change the % to something else and ditch the s so that there is less to think about?

We can change from the named object syntax `%(username)s to the`%s`, but one nice property is for localizers is that these are self documenting.

Would you rather translate

```
Persona lets you use your %s account to sign into sites like %s.
```

or 

```
Persona lets you use your %(emailProvider)s account to sign into sites like %(aWebsite)s.
```

We [don't support gettext comments](https://github.com/mozilla/browserid/issues/1132) yet. So localizers have no context what the string is about. At least `emailProvider` and `aWebsite` is some documentation to help them.

**Comment by ozten** [comment]
> I updated lockdown.json so that the Travis tests pass.

Awesome! Sorry about that.

**Comment by ozten** [comment]
> I tried running
> CONFIG_FILES=./config/production.json node ./scripts/check_l10n_config.js 

This is not a new script. This works "as well" as it does today. I'm not sure anyone uses this script.

**Comment by ozten** [comment]
Good call on missing references to creating JSON files.

I've updated docs/I18N.md with developer focused notes.

Neither `dev` nor this pull request use MO files. That was changed several months ago.

This pull request doesn't introduce any new tooling. This is the same flow we currently do.

**Comment by shane-tomlinson** [comment]
> Neither dev nor this pull request use MO files. That was changed several months ago.

There is a script in the scripts directory called `compile_mo.sh`, is this still needed? There is also `check_po.sh` and `extract_po.sh`.

I have created a helper script that pulls the svn repo, compiles the json files, and compresses the templates based on this update, but I am not sure whether this is the right place to put this. Is there value in adding it to this PR?

**Comment by shane-tomlinson** [comment]
I am comfortable with this this PR. Tested locally and with ephemeral instances. 

r+, will merge when given the green light that the repo is open.

@mathjazz - the updates in this PR are going to cause a lot of fuzzy mis-matches in the locales due to the new `s`'s found in EJS variables.

For example:
`%(variableName)` is now `%(variableName)s` so that our scheme stays consistent with the rest of Mozilla.

**Comment by mathjazz** [comment]
Thanks, @shane-tomlinson.

I think I talked to @ozten about this some time ago, so we can use a script to replace the variable names.

**Pull request merged**